frontend/src/selectors.tsx

```diff
@@ -15,3 +29,46 @@ export const clusterCuratorTemplatesValue = selector({
         )
     },
 })
+
+const basicCurations: Curation[] = ['install', 'upgrade']
+const allCurations: Curation[] = [...basicCurations, 'scale', 'destroy']
+export const clusterCuratorSupportedCurationsValue = selector({
+    key: 'clusterCuratorSupportedCurations',
+    get: ({ get }) => {
+        const settings = get(settingsState)
+        return settings.ansibleIntegration === 'enabled' ? allCurations : basicCurations
+    },
+})
+
+export const providerConnectionsValue = selector({
+    key: 'providerConnections',
+    get: ({ get }) => {
+        const secrets = get(secretsState)
+        return secrets.map(unpackProviderConnection)
+    },
+})
+
+export const validClusterCuratorTemplatesValue = selector({
```

Comment:
@RobDolares This selector may be useful for you in implementing the batch update. It gives you all the templates that don't refer to any missing Ansible secrets.